internal/node/indexer/server.go

```diff
@@ -42,8 +42,8 @@ type Server struct {
 
 func (s *Server) Run(ctx context.Context) error {
 	var (
-		// TODO Develop a more effective solution to implement back pressure.
-		tasksChan = make(chan []engine.Task, 1)
+		// TODO Develop a more effective solution to implement back pressure, and stores source state in a task group.
```

Comment:
I plan to completely remove `State` from `engine.Source`, and reimplement it in `engine.Task`. This plan can avoid strong coupling between `checkpoint` and `source`, as well as implement a well-rounded cached task queue.